tiny_dnn/activations/elu_layer.h

```diff
@@ -33,14 +50,15 @@ class elu_layer : public activation_layer {
                            const vec_t &dy) override {
     for (size_t j = 0; j < x.size(); j++) {
       // dx = dy * (gradient of elu)
-      dx[j] = dy[j] * (y[j] > float_t(0) ? float_t(1) : (float_t(1) + y[j]));
+      dx[j] = dy[j] * (y[j] > float_t(0) ? float_t(1) : (alpha_ + y[j]));
```

Comment:
It seems correct according to below references.

https://arxiv.org/pdf/1511.07289.pdf
https://en.wikipedia.org/wiki/Activation_function#Comparison_of_activation_functions

Chainer's `ELU::backward_cpu` looks different though.
https://github.com/chainer/chainer/blob/master/chainer/functions/activation/elu.py
It multiplies with alpha. I don't know why...

Caffe's `ELULayer::Backward_cpu` uses addition, so it looks the same.
https://github.com/BVLC/caffe/blob/master/src/caffe/layers/elu_layer.cpp

TensorFlow's `EluGradOp::Compile` does not use alpha.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/tf2xla/kernels/elu_op.cc#L49

After checking this gradient code, I wonder if `selu_layer::backward_activation` is OK.

Reply:
Why would they change the backward pass and keep forward pass the same? :confused:

Reply:
I referred to the paper